src/MulticastService.cs

```diff
@@ -271,9 +310,10 @@ public void Stop()
             QueryReceived = null;
             AnswerReceived = null;
             NetworkInterfaceDiscovered = null;
-            if (listenerCancellation != null)
+            using (var lc = listenerCancellation)
             {
-                listenerCancellation.Cancel();
+                listenerCancellation = null;
+                lc?.Cancel();
```

Comment:
Very odd code!  What about this

```
if (lisenterCancellation != null)
{
  var lc = listenerCancellation;
  listenerCancellation = null;
  lc.Cancel();
  lc.Dispose();
}
```

Reply:
You are doing the exact same thing as the compiler would otherwise do for you.
The "using" statement creates a local reference to you disposable object and dispose it when the block exits. I can therefor null the 'listenerCancellation' before calling cancel. In a multithreaded environment this is a good thing, as this makes sure we do not have a reference to a canceled token source. So we only need to check for null when using it.

Anyway, writing this I double checked the code, I spotted an issue in the PollNetworkInterfaces() method. I will fix that as well.

I'll leave this for now, but let me know if you want it changed :)